src/exo0 - Composing with pipe and flow/exo0.effect.exercise.ts

```diff
@@ -0,0 +1,81 @@
+// Effect training introduction
+// Composing computations with `pipe` and `flow`
+
+// Functional programming is all about composing small functions together like
+// Lego bricks to build more and more complex computations.
+//
+// Strictly speaking, composing two functions `f` and `g` means applying the
+// result of the first one to the second one. By applying this composition
+// over and over you can chain multiple functions together.
+//
+// The `Effect` library provides helpers to do that:
+// - `pipe` which first needs to be fed a value to start the pipe and then
+//   any number of functions to be applied sequentially.
+// - `flow` which is the same thing but where we do not have to provide the
+//   first value. It will then return a function which will expect that value
+//   to be provided
+//
+// ```ts
+// flow(f, g, h) === x => pipe(x, f, g, h)
+// pipe(x, f, g, h) === flow(f, g, h)(x)
+// ```
+//
+// NOTES:
+// - `flow(f) === f`
+// - `pipe(x, f) === f(x)`
+
+// You'll need these imports to solve the exercise:
+import { pipe, flow } from "effect"
+
+export const isEven = (value: number) => value % 2 === 0
+
+export const not = (value: boolean) => !value
+
+///////////////////////////////////////////////////////////////////////////////
+//                               PIPE & FLOW                                   //
+///////////////////////////////////////////////////////////////////////////////
+
+// For this exercise each function needs to be implemented using both forms:
+// - a `pipe` implementation (P suffix)
+// - a `flow` implementation (F suffix)
+
+// Using only the two helpers `isEven` and `not` at the top of this file (and
+// `pipe` or `flow`), write the function `isOdd` that checks if a number is
+// odd.
+
+export const isOddP = (value: number) => pipe(value, isEven, not)
```

Comment:
Seems like you solve the exercise in the file where you're supposed to keep `unimplemented` methods! :)